packages/flutter/lib/src/material/ink_decoration.dart

```diff
@@ -252,16 +252,21 @@ class _InkState extends State<Ink> {
   }
 
   Widget _build(BuildContext context, BoxConstraints constraints) {
+    // if _ink have IndexedStack ancestor(s) and not on the display path,
+    // _ink should not paint.
+    final bool inkNeedsPaint = IndexedStack.addIndexedStackScopeAncestorsDependencies(context);
```

Comment:
This seems like a bandage fix, because it only applies to indexed stack. I believe there are more widgets that may not paint all of their children. Further more, you may run into circular dependencies if that widget is in Cupertino library.

Reply:
I feel the same way. I'll see if there's a better plan.